packages/amplify_authenticator/lib/src/utils/validators.dart

```diff
@@ -17,6 +17,7 @@ import 'package:amplify_authenticator/amplify_authenticator.dart';
 import 'package:amplify_flutter/amplify_flutter.dart';
 import 'package:flutter/material.dart';
 
+final usernameRegex = RegExp(r'[\p{L}\p{M}\p{S}\p{N}\p{P}]+');
```

Comment:
Unfortunately it seems that the flavor of regex that Dart uses does not support `\p{L}`, `\p{M}`, etc. `\p{L}` is treated two separate matches - `\p` (literal "p" character) and `{L}` ("L" character). So in Dart, `[\p{L}\p{M}\p{S}\p{N}\p{P}]+` will match and string that has one of the follow characters in it - pLMSNP.

You can see the difference between common flavors on https://regex101.com/. PCRE2 allows for these characters. Ecmascript, which I believe is what dart follows, does not.

I propose we use `RegExp(r'^\S+$');` in this PR, and I can open a separate issue to look into a more complete solution. `RegExp(r'^\S+$');` will allow for any non white space character, which will be an improvement on the current validator which allows for usernames with white space. `RegExp(r'^\S+$');` will still allow for some characters which Cognito will reject, such as emojis, but this is still a step in the right direction.

In order to align this completely with Cognito, we will either need an exhaustive list of all the characters that are included in `[\p{L}\p{M}\p{S}\p{N}\p{P}]` in regex flavors that allow for this, or possibly a 3rd party lib.

This may apply to other validators besides username. We use `RegExp(r'^\S+@\S+$')` for email validation, which is probably also allowing for some characters that Cognito will reject, such as emojis.

@dnys1 - any objections to using `RegExp(r'^\S+$');` for now?

Reply:
Good catch. 

> any objections to using RegExp(r'^\S+$'); for now?
Works for me - maybe with a `TODO` comment as well.

Reply:
I've updated the regex expression to match the one you agreed on and added the required `TODO`.
It now looks like this:
```dart
/* TODO: update `usernameRegex` expression to match Cognito requirements
* Proposed expression `[\p{L}\p{M}\p{S}\p{N}\p{P}]+` does not work as 
* expected due to a mismatch in Regex flavor used by dart
*/
final usernameRegex = RegExp(r'^\S+$');
```